tests/datadriven/test_reads.py

```diff
@@ -67,6 +71,25 @@ def getDataModelClass(self):
     def getProtocolClass(self):
         return protocol.ReadGroupSet
 
+    def testPrograms(self):
+        # test that program info is set correctly
+        readGroupSet = self._gaObject
+        for readGroup in readGroupSet.getReadGroups():
+            readGroupInfo = self._readGroupInfos[readGroup.getSamFilePath()]
+            gaPrograms = readGroup.getPrograms()
+            htslibPrograms = readGroupInfo.programs
```

Comment:
Need to check the lengths here. If (say) gaPrograms was empty here, this would still pass happily as zip doesn't care if the lists are different lengths.